src/java.base/share/classes/java/lang/invoke/X-VarHandleSegmentView.java.template

```diff
@@ -33,723 +33,721 @@ import static java.lang.invoke.SegmentVarHandle.*;
 
 #warn
 
-final class VarHandleSegmentAs$Type$s {
+{#if[byte]?final:sealed} class VarHandleSegmentAs$Type$s {
 
 #if[!byte]
     static final int NON_PLAIN_ACCESS_MIN_ALIGN_MASK = $BoxType$.BYTES - 1;
 
 #end[byte]
-    static VarForm selectForm(long alignmentMask, boolean noStride) {
+    static VarForm selectForm(long alignmentMask, boolean constantOffset) {
 #if[byte]
-        return noStride ? Unaligned.FORM : Unaligned.OFFSET_FORM;
+        return constantOffset ? CONSTANT_OFFSET_FORM : TAKE_OFFSET_FORM;
 #else[byte]
         return (alignmentMask & NON_PLAIN_ACCESS_MIN_ALIGN_MASK) != NON_PLAIN_ACCESS_MIN_ALIGN_MASK ?
-                (noStride ? Unaligned.FORM : Unaligned.OFFSET_FORM) :
-                (noStride ? Aligned.FORM : Aligned.OFFSET_FORM);
+                (constantOffset ? CONSTANT_OFFSET_FORM : TAKE_OFFSET_FORM) :
+                (constantOffset ? VarHandleSegmentAs$Type$sAligned.CONSTANT_OFFSET_FORM : VarHandleSegmentAs$Type$sAligned.TAKE_OFFSET_FORM);
 #end[byte]
     }
 
-    static {#if[byte]?final:sealed} class Unaligned {
+    static final VarForm CONSTANT_OFFSET_FORM = new VarForm(VarHandleSegmentAs$Type$s.class, MemorySegment.class, $type$.class, long.class);
+    static final VarForm TAKE_OFFSET_FORM = new VarForm(VarHandleSegmentAs$Type$s.class, MemorySegment.class, $type$.class, long.class, long.class);
```

Comment:
what does TAKE mean here? (the old names weren't great either, I realize). I think it would be better to decide, once and for all, how to refer to these things, and then use consistent naming throughout. I see we have discussed using "noStride" vs. "fixedOffset". Since the implementation now uses the term "constantOffset", let's just call the other "variableOffset" ?